src/web_interface/components/plugin_routes.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+import importlib
+import inspect
+import pkgutil
+
+from helperFunctions.fileSystem import get_src_dir
+from web_interface.components.component_base import ComponentBase
+
+
+ROUTES_MODULE_NAME = 'routes'
+PLUGIN_CATEGORIES = ['analysis', 'compare', 'unpacking']
+PLUGIN_DIR = '{}/plugins'.format(get_src_dir())
+
+
+class PluginRoutes(ComponentBase):
+    def _init_component(self):
+        plugin_list = self._find_plugins()
+        self._register_all_plugin_endpoints(plugin_list)
+
+    def _register_all_plugin_endpoints(self, plugins_by_category):
+        for plugin_type, plugin_list in plugins_by_category:
+            for plugin in plugin_list:
+                if self._module_has_routes(plugin, plugin_type):
+                    self._import_module_routes(plugin, plugin_type)
+
+    def _find_plugins(self):
+        plugin_list = []
+        for plugin_category in PLUGIN_CATEGORIES:
+            plugin_list.append((plugin_category, self._get_modules_in_path('{}/{}'.format(PLUGIN_DIR, plugin_category))))
+        return plugin_list
+
+    def _module_has_routes(self, plugin, plugin_type):
+        plugin_components = self._get_modules_in_path('{}/{}/{}'.format(PLUGIN_DIR, plugin_type, plugin))
+        return ROUTES_MODULE_NAME in plugin_components
+
+    def _import_module_routes(self, plugin, plugin_type):
+        module = importlib.import_module('plugins.{0}.{1}.{2}.{2}'.format(plugin_type, plugin, ROUTES_MODULE_NAME))
+        if hasattr(module, 'PluginRoutes'):
+            module.PluginRoutes(self._app, self._config)
+        for rest_class in [getattr(module, attribute) for attribute in dir(module) if 'Rest' in attribute and inspect.isclass(getattr(module, attribute))]:
```

Comment:
That's super ugly ^^

Reply:
And also not the way you should do it..
Rather you should test if a child of `flask_restful.Resource` is found.